### DISCUSSION TITLE
pag动画开发，是否有设置播放动画速率的方法

### DISCUSSION THREAD

**Discussion by 1169573997** [Tencent/libpag#2332] [Q&A] [answered]
开发过程中，有一个需求，比如，导出的动画时长是5s，但是在程序里面想要这个动画时长在3s内播放完，就想着能不能设置一下播放的速率，找了一圈都没找到方法，请求有大佬提供方法吗？

自己也写了一个定时器来逐步播放帧，但是播放出来的效果并不理想。

**Comment by kevingpqi123** [comment, marked as answer]
这些基础的问题建议看官方文档：
https://pag.art/docs/time-stretch.html

以及系统看 API 接口
Set the duration of this PAGFile. Passing a value less than or equal to 0 resets the duration to its default value.

- (void)setDuration:(int64_t)duration
Declared In
PAGFile.h

**Reply by 1169573997** [reply]
文档里面的方法都试了， 设置startTime  设置duration 之类的。setprogress 都尝试了一下，还是不能达到我要的需求。
文件动画是已经由设计师从ae里面导出来了。文件动画的本身时长已经固定， 就是想要达到由代码来控制这个动画播放时长。
duration 这个方法已经试过 他只能设置一个播放时长 而不能改变播放动画的速率

需求就类似与lottie动画里面的  public func play(
    fromProgress: AnimationProgressTime? = nil,
    toProgress: AnimationProgressTime,
    loopMode: LottieLoopMode? = nil,
    completion: LottieCompletionBlock? = nil)
  方法和      public var animationSpeed: CGFloat = 1  属性的搭配使用

**Reply by kevingpqi123** [reply]
https://pag.art/docs/time-stretch.html
这个文档为啥不看呢，你只设置一个肯定没有用，这个文档看明白，让设计师设置下伸缩区间重新导出

Set the [timeStretchMode](https://pag.art/apis/ios/Classes/PAGFile.html#//api/name/timeStretchMode) of this file.

- (void)seTimeStretchMode:(PAGTimeStretchMode)value
Declared In
PAGFile.h

**Reply by 1169573997** [reply]
好的，感谢大佬解惑。非常感谢
